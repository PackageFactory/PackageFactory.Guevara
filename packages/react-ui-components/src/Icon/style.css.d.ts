export interface IStyleCss {
  'icon': string;
  'icon--paddedLeft': string;
  'icon--paddedRight': string;
  'icon--color-warn': string;
  'icon--color-error': string;
  'icon--color-primaryBlue': string;
}

export const locals: IStyleCss;
