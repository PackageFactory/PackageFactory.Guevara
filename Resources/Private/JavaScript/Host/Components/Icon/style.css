.icon {
  display: inline-block;
  font: normal normal normal 14px/1 FontAwesome;
  font-size: inherit;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
.icon--big {
  font-size: 1.5em;
}
.icon--small {
  font-size: .75em;
}
.icon--tiny {
  font-size: .5em;
}
.icon--paddedLeft {
    padding-left: .75em;
}
.icon--paddedRight {
    padding-right: .75em;
}
