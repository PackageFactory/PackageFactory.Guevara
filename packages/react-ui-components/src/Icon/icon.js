import React from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import mapper from './mapper';

const Icon = props => {
    const {size, padded, theme, label} = props;
    const iconClassName = props.icon;
    const classNames = mergeClassNames({
        [theme.icon]: true,
        [iconClassName]: true,
        [props.className]: props.className && props.className.length,
        [theme['icon--big']]: size === 'big',
        [theme['icon--medium']]: size === 'medium',
        [theme['icon--small']]: size === 'small',
        [theme['icon--tiny']]: size === 'tiny',
        [theme['icon--paddedLeft']]: padded === 'left',
        [theme['icon--paddedRight']]: padded === 'right',
        [theme['icon--spin']]: props.spin
    });

    const mappedIcon = mapper(props.icon);
    const iconArray = mappedIcon.split(' ');
    let icon = mappedIcon;
    let prefix = 'fas';
    if (iconArray.length > 1) {
        prefix = iconArray[0];
        const iconClass = iconArray[1];
        if (iconClass.startsWith('fa-')) {
            icon = iconClass.substr(3);
        } else {
            icon = iconClass;
        }
    }

    return <FontAwesomeIcon icon={[prefix, icon] || 'question'} aria-label={label} className={classNames} />;
};
Icon.propTypes = {
    /**
     * The ID of the icon to render.
     */
    icon: PropTypes.string,

    /**
     * The (accessibility) label for this icon
     */
    label: PropTypes.string,

    /**
     * Controls the rendered size of the icon.
     */
    size: PropTypes.oneOf(['big', 'medium', 'small', 'tiny']),

    /**
     * Controls the padding around the icon in a standardized way.
     */
    padded: PropTypes.oneOf(['none', 'left', 'right']),

    /**
     * An optional `className` to attach to the wrapper.
     */
    className: PropTypes.string,

    /**
     * When truthy, the icon will spin continously.
     */
    spin: PropTypes.bool,

    /**
    * An optional css theme to be injected.
    */
    theme: PropTypes.shape({
        'icon': PropTypes.string,
        'icon--big': PropTypes.string,
        'icon--small': PropTypes.string,
        'icon--tiny': PropTypes.string,
        'icon--paddedLeft': PropTypes.string,
        'icon--paddedRight': PropTypes.string,
        'icon--spin': PropTypes.string
    }).isRequired
};

export default Icon;
