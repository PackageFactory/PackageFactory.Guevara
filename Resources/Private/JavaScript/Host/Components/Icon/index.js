import React, {PropTypes} from 'react';
import mergeClassNames from 'classnames';
import {service} from 'Shared/';
import style from './style.css';
import iconStyles from './icons.css';
import {validateIconId} from './IconNames.js';
const {logger} = service;

const getIconClassName = id => {
    const {iconName} = validateIconId(id);

    return iconStyles[iconName];
};

const Icon = props => {
    const {size, padded} = props;
    const iconClassName = getIconClassName(props.icon);
    const classNames = mergeClassNames({
        [style.icon]: true,
        [iconClassName]: true,
        [props.className]: props.className && props.className.length,
        [style['icon--big']]: size === 'big',
        [style['icon--small']]: size === 'small',
        [style['icon--tiny']]: size === 'tiny',
        [style['icon--paddedLeft']]: padded === 'left',
        [style['icon--paddedRight']]: padded === 'right'
    });

    return (
        <i className={classNames}></i>
    );
};
Icon.propTypes = {
    // The icon key of Font-Awesome.
    icon(props, propName) {
        const id = props[propName];
        const {isValid, isMigrationNeeded, iconName} = validateIconId(id);

        if (!isValid) {
            if (isMigrationNeeded && iconName) {
                logger.warn(`Font-Awesome has been updated. The icon name "${id}" has been renamed.

Please adjust the icon configurations in your .yaml files to the new icon name "${iconName}".

https://github.com/FortAwesome/Font-Awesome/wiki/Upgrading-from-3.2.1-to-4`);
            } else {
                return new Error(`Icon name "${id}" was not a found in Font-Awesome 4.5.
Please use the icon names from the Font-Awesome website.

http://fortawesome.github.io/Font-Awesome/icons/`);
            }
        }
    },

    // Style related propTypes.
    size: PropTypes.oneOf(['big', 'small', 'tiny']),
    padded: PropTypes.oneOf(['none', 'left', 'right']),
    className: PropTypes.string
};

export default Icon;
