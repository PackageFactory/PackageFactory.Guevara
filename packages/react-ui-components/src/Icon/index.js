import fontawesome from '@fortawesome/fontawesome';
import brands from '@fortawesome/fontawesome-free-brands';
import solid from '@fortawesome/fontawesome-free-solid';
import regular from '@fortawesome/fontawesome-free-regular';
import {themr} from 'react-css-themr';
import identifiers from './../identifiers.js';
import style from './style.css';
import Icon from './icon.js';

fontawesome.library.add(brands, solid, regular);

const ThemedIcon = themr(identifiers.icon, style)(Icon);

export default ThemedIcon;
