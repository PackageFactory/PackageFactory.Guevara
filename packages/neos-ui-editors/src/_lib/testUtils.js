import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
