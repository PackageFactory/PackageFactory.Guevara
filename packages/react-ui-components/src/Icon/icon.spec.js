import sinon from 'sinon';
import {createShallowRenderer} from './../_lib/testUtils.js';
import Icon, {iconPropValidator} from './icon.js';

const defaultProps = {
    theme: {},
    iconMap: {},
    _makeValidateId: () => id => ({isValid: true, isMigrationNeeded: false, iconName: id}),
    _makeGetClassName: () => id => id,
    onDeprecate: () => null
};
const shallow = createShallowRenderer(Icon, defaultProps);

test('should render a "i" node.', () => {
    const tag = shallow();

    expect(tag.type()).toBe('i');
});
test('should add the passed "className" prop to the rendered node if passed.', () => {
    const tag = shallow({className: 'testClassName'});

    expect(tag.hasClass('testClassName')).toBeTruthy();
});
test('should call the "fontAwesome.getClassName" api method and render the returned className.', () => {
    const tag = shallow({icon: 'fooIconClassName'});

    expect(tag.hasClass('fooIconClassName')).toBeTruthy();
});

test('iconPropValidator() should call the "onDeprecate" in case a migration is needed.', () => {
    const props = {
        icon: 'fooIconClassName',
        _makeValidateId: () => id => ({isValid: false, isMigrationNeeded: true, iconName: id}),
        onDeprecate: sinon.spy()
    };

    iconPropValidator(props, 'icon');

    expect(props.onDeprecate.calledOnce).toBeTruthy();
});
test('iconPropValidator() should not call the "onDeprecate" multiple times for the same icon id.', () => {
    const props = {
        icon: 'barIconClassName',
        _makeValidateId: () => id => ({isValid: false, isMigrationNeeded: true, iconName: id}),
        onDeprecate: sinon.spy()
    };

    iconPropValidator(props, 'icon');
    iconPropValidator(props, 'icon');

    expect(props.onDeprecate.callCount).toBe(1);
});
test('iconPropValidator() should return an error if the iconName was not found.', () => {
    const props = {
        icon: 'bazIconClassName',
        _makeValidateId: () => () => ({isValid: false, isMigrationNeeded: false, iconName: null})
    };
    const result = iconPropValidator(props, 'icon');

    expect(result instanceof Error).toBeTruthy();
});
test('iconPropValidator() should not return an error if no condition was matched.', () => {
    const props = {
        icon: 'bazIconClassName',
        _makeValidateId: () => () => ({})
    };
    const result = iconPropValidator(props, 'icon');

    expect(result).toBe(undefined);
});
