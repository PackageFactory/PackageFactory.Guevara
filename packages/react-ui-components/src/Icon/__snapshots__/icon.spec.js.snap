// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> should render correctly. 1`] = `
<FontAwesomeIcon
  _makeGetClassName={[Function]}
  _makeValidateId={[Function]}
  border={false}
  className="fooIconClassName"
  fixedWidth={false}
  flip={null}
  icon={
    Array [
      "fas",
      "fooIconClassName",
    ]
  }
  iconMap={Object {}}
  inverse={false}
  listItem={false}
  mask={null}
  onDeprecate={[Function]}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  symbol={false}
  title=""
  transform={null}
/>
`;
